banyan/src/stream_builder.rs

```diff
@@ -158,6 +170,21 @@ impl<T: TreeTypes> StreamBuilder<T> {
         self.root.as_ref().map(|x| x.as_ref())
     }
 
+    /// Modify a StreamBuilder and roll back the changes if the operation was not successful
+    ///
+    /// Note that consumed offets are *not* rolled back to make sure we don't reuse offsets.
```

Comment:
Does that mean that there might be offset gaps now within the tree? So a chunk is 100..200 but only `data.len()` is 50?

Reply:
yeah, I was similarly confused: these are different offsets (namely bytes written with the stream cipher) — these new offsets should definitely be named differently

Reply:
I see, that makes sense.

:+1: for renaming. I saw `StreamOffset` somewhere? Maybe `CipherOffset`/`CryptoOffset`/`EncryptionOffset`? `CipherPointer`?

Reply:
Yes, you can have gaps. When you build a tree you won't have any gaps, but as soon as you do things like pack or retain, there will be gaps. Nothing wrong with that though, you are not going to run out of u64 any time soon, and gaps are not a problem.

`CIpherOffset` sounds reasonable. Will rename.

Reply:
… and if someone is about to run out of u64 cipher offsets, we can rekey the tree. But we need to hurry, at 100MB/s we only have about 5700 years before this happens 😀.

Reply:
I was going to add methods for rekeying in any case, but it is pretty straightforward and not that urgent...